DashAI/back/preprocessor/column_dropper_by_index.py

```diff
@@ -0,0 +1,55 @@
+from beartype import beartype
+from datasets import DatasetDict
+
+from DashAI.back.dataloaders.classes.dashai_dataset import DashAIDataset
+from DashAI.back.preprocessor.base_converter import BaseConverter
+
+
+class ColumnDropperByIndex(BaseConverter):
+    """Converter to drop columns from the dataset by column index"""
+
+    @beartype
+    def __init__(self, columns: tuple[int, int]):
+        """Constructor with columns to be dropped by column index
+
+        Parameters
+        ----------
+        columns : tuple[int, int]
+            Columns to be dropped. The tuple contains the start and end index of the
+            columns to be dropped.
+        """
+        self.columns = columns
+
+    @beartype
+    def fit(self, dataset: DatasetDict) -> BaseConverter:
```

Comment:
It seems to me that the correct thing to do here would be to return Type["BaseConverter"] (type should be imported from typing).

```suggestion
    def fit(self, dataset: DatasetDict) -> Type["BaseConverter"]:
```